src/Observable.js

```diff
@@ -315,6 +315,26 @@ export class Observable {
     }));
   }
 
+  async all() {
+    let promise = new Promise((resolve, reject) => {
+      this
+        .reduce((previous, current) => {
+          previous.push(current);
+          return previous;
+        }, [])
+        .subscribe(o => resolve(o), e => reject(e));
+    });
+
+    return promise;
+  }
+
+  async *[Symbol.asyncIterator]() {
```

Comment:
With `all`, can we just do this?

```js
for (const value of await observable.all()) {
  // Loop body...
}
```

I'd like to defer adding `Symbol.asyncIterator` because of the buffering issues.